compiler/test/fail_compilation/test20956.d

```diff
@@ -0,0 +1,19 @@
+/* REQUIRED_ARGS: -preview=dip1000
+TEST_OUTPUT:
+---
+fail_compilation/test20956.d(105): Error: function `test20956.forwardDg` cannot close over `ref` variable `c`
+---
+ */
+
+// https://issues.dlang.org/show_bug.cgi?id=20956
+
+#line 100
+
+@safe:
+
+alias DG = void delegate() @safe;
+
+DG forwardDg(ref int c)
+{
+    return () {assert(c == 42);};
+}
```

Comment:
Shouldn't the compiler infer the closure to be `scope` in this case ?
Regardless if it does/can infer it currently, we should not have this error for `scope` delegate, should we ?

Reply:
`ref` parameters are implicitly pointers, and are automatically not allowed to escape the function. No need for `scope`.

The trouble here is the delegate escapes the function, thereby escaping the reference to c.

Reply:
Right, we're returning the `delegate`. Can we add a test that not escaping the delegate (the delegate being `scope`) does not trigger error ?

Reply:
The error is triggered when the delegate is tested to see if a dynamic closure is being created. So the tests for @nogc already do this.